reana_server/reana_admin/cli.py

```diff
@@ -638,24 +638,46 @@ def set_quota_limit(
 
 @reana_admin.command(
     "quota-set-default-limits",
-    help="Set default quota limits to users that do not have any.",
+    help="Set default quota limits for users who do not have any custom limits defined.",
 )
 @admin_access_token_option
 @click.pass_context
 def set_default_quota_limit(ctx, admin_access_token: str):
-    """Set default quota limits to users that do not have any."""
-    users_without_quota_limits = User.query.filter(~User.resources.any()).all()
+    """Set default quota limits for users who do not have any custom limits defined."""
+    users_without_quota_limits = (
+        Session.query(User)
+        .filter(
+            User.id_.in_(
+                Session.query(UserResource.user_id).filter(
+                    UserResource.quota_limit == 0
+                )
+            )
+        )
+        .all()
+    )
+
     if not users_without_quota_limits:
         click.secho("There are no users without quota limits.", fg="green")
         sys.exit(0)
-    for resource in Resource.query:
-        ctx.invoke(
-            set_quota_limit,
-            emails=[user.email for user in users_without_quota_limits],
-            resource_name=resource.name,
-            resource_type=resource.type_.name,
-            limit=DEFAULT_QUOTA_LIMITS.get(resource.type_.name),
-        )
+
+    resources = Resource.query.all()
+
+    for user in users_without_quota_limits:
+        for resource in resources:
+            user_resource = UserResource.query.filter_by(
+                user_id=user.id_, resource_id=resource.id_
+            ).first()
+
+            if user_resource and user_resource.quota_limit == 0:
+                # If no limit exists, set the default limit
+                ctx.invoke(
+                    set_quota_limit,
+                    emails=[user.email],
+                    resource_name=resource.name,
+                    resource_type=resource.type_.name,
+                    limit=DEFAULT_QUOTA_LIMITS.get(resource.type_.name),
```

Comment:
Works nicely, but perhaps we should check whether the setting is actually needed here, because otherwise some unnecessary operations would be called:

```
Quota limit 0 for 'cpu (processing time)' successfully set to users ['aaa@example.org'].                                       │
Quota limit 0 for 'cpu (processing time)' successfully set to users ['bbb@example.org'].
```

This populates the table with zeros which may not be necessary.  (Even though later upgrade to a real value would well.)